registry/v2/registry.go

```diff
@@ -480,14 +477,21 @@ func (r *registry) UploadProgress(ctx echo.Context) error {
 	namespace := ctx.Param("username") + "/" + ctx.Param("imagename")
 	uuid := ctx.Param("uuid")
 
-	skylink, err := r.localCache.GetSkynetURL(namespace, uuid)
+	skylink, err := r.store.GetContentHashById(ctx.Request().Context(), uuid)
 	if err != nil {
-		locationHeader := fmt.Sprintf("/v2/%s/blobs/uploads/%s", namespace, uuid)
-		ctx.Response().Header().Set("Location", locationHeader)
-		ctx.Response().Header().Set("Range", "bytes=0-0")
-		ctx.Response().Header().Set("Docker-Upload-UUID", uuid)
-		r.logger.Log(ctx, err)
-		return ctx.NoContent(http.StatusNoContent)
+		errMsg := r.errorResponse(RegistryErrorCodeBlobUnknown, err.Error(), nil)
+		r.logger.Log(ctx, fmt.Errorf("%s", errMsg))
+		return ctx.JSONBlob(http.StatusNotFound, errMsg)
+	}
+
+	if skylink == "" {
+		detail := map[string]interface{}{
+			"error": "skylink is empty",
+		}
+		e := fmt.Errorf("skylink is empty").Error()
```

Comment:
@guacamole please remove this, since we're already doing this is detail map above